services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -576,5 +588,27 @@ async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
                         raise _RetryError() from e
                     log.debug(
                         "Skip update for service '%s' which could not be found",
-                        scheduler_data.service_name,
+                        service_name,
                     )
+
+
+async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
+    def _update_labels_in_spec(spec: dict) -> dict:
+        spec["Labels"][
+            DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL
+        ] = scheduler_data.as_label_data()
+        return spec
+
+    await _update_service_spec(
+        service_name=scheduler_data.service_name,
+        update_spec_data=_update_labels_in_spec,
+    )
+
+
+async def constrain_service_to_node(service_name: str, node_id: str) -> None:
+    def _add_placement_constraint(spec: dict) -> dict:
```

Comment:
if I understand correctly all these calls are done in sequence right? I start to wonder if the docker engine is very happy about getting calls every few milliseconds and if that might have some influence on its responsiveness. only a thought for now...

Reply:
So all the calls are sequenced for each dynamic-sidecar. Generally speaking, unless for some bug, there is no "hammering" the docker engine with requests.

We might still want to introduce a small `jitter/delay` when starting a study with a lot of dynamic-sidecars.
For now, the docker-engine will get a huge spike of requests, since all of them are started together in parallel.